client/my-sites/earn/memberships/index.jsx

```diff
@@ -680,8 +680,8 @@ class MembershipsSection extends Component {
 			<div>
 				<QueryMembershipsEarnings siteId={ this.props.siteId } />
 				<QueryMembershipsSettings siteId={ this.props.siteId } source={ this.props.source } />
-				{ this.props.connectedAccountId && this.renderStripeConnected() }
-				{ this.props.connectUrl && ! this.props.connectedAccountId && this.renderConnectStripe() }
+				{ ! this.props.connectUrl && this.props.connectedAccountId && this.renderStripeConnected() }
```

Comment:
I see. [We are passing](https://code.a8c.com/source/wpcom-git/browse/trunk/wp-content/lib/memberships/connected-accounts.php$160) `connectedAccountId` even if the account is not enabled. 
That is why this logic seems complex for what it is... Maybe we should change this behaviour in the backend (do not pass account-id if not enabled?) instead of doing this? I'll leave this up to you...

Reply:
I responded in D114449-code; I have no strong feelings about this.

Reply:
> I responded in D114449-code; I have no strong feelings about this.

Let's fix it in the back end , then

>  I think we would still need to make changes on the frontend regardless, right?

I don't think we would need this if we don't send the `connected_account_id` when `connect_url` is sent.

Reply:
I could be misunderstanding things here, but if I remember correctly, the code on trunk is:

```
				{ this.props.connectedAccountId && this.renderStripeConnected() }
				{ this.props.connectUrl && ! this.props.connectedAccountId && this.renderConnectStripe() }

				{  ! this.props.connectedAccountId && ! this.props.connectUrl && (
					<div className="earn__payments-loading">
						<LoadingEllipsis />
					</div>
				) }
```

If `this.props.connectedAccountId`  and `this.props.connectUrl ` is falsy then the `LoadingEllipsis` will be displayed. If I understand this correctly, when Stripe is disconnected, the aim is to send an empty string for `connectedAccountId` and `connectUrl` which will cause the `LoadingEllipsis` to be loaded indefinitely?

Reply:
> If `this.props.connectedAccountId` and `this.props.connectUrl ` is falsy then the `LoadingEllipsis` will be displayed. If I understand this correctly, when Stripe is disconnected, the aim is to send an empty string for `connectedAccountId` and `connectUrl` which will cause the `LoadingEllipsis` to be loaded indefinitely?

when Stripe is disconnected, `connectUrl()` will be sent back NOT empty. As it is done at the moment I believe...

Reply:
Oh right! I went ahead and added your feedback to D114449-code. The change didn't appear to fix https://github.com/Automattic/wp-calypso/issues/77898 🤔 

Another observation, we could also simplify the frontend logic to use only the `connectUrl ` since using `connectedAccountId` and `connectUrl` may not be needed to gate the UI?

```
	{ ! this.props.connectUrl && this.renderStripeConnected() }
	{ this.props.connectUrl && this.renderConnectStripe() }

	{  ! this.props.connectedAccountId && ! this.props.connectUrl  && (
		<div className="earn__payments-loading">
			<LoadingEllipsis />
		</div>
       ) }
```

Reply:
Additional context can be found on slack: p1689019725361129/1689003001.792249-slack-C01B6KEJ5GE